Adds pyroscope panel and datasource plugins

petethepig opened the pull request:
Hi Grafana team!

This PR adds [pyroscope](https://github.com/pyroscope-io/pyroscope/) panel and datasource plugins.

To test the datasource plugin you can run `docker run -it -p 4040:4040 pyroscope/pyroscope:latest server` and use localhost:4040 as the address in datasource config. For app name you can use `pyroscope.server.cpu` and that be pyroscope profiling itself. We also have a publicly available server at https://demo.pyroscope.io/ .

We also have a live demo deployed at [grafana-demo.pyroscope.io](https://grafana-demo.pyroscope.io/d/9_aiZ46Gz/pyroscope-plugins?orgId=1)

Both plugins are currently not signed, I'll update this PR once we get our grafana account approved. The name of the Grafana Cloud org is `pyroscopedmitry`.

marcusolsson commented:
Hi, and thanks for your submission!

Unfortunately, I can't seem to access the plugin URL. Have you published the releases yet?

petethepig commented:
@marcusolsson Hi Marcus. My bad, the releases were in "draft" state. I fixed it now. Let us know when we can sign the plugins and I will update the PR with signed versions.

marcusolsson commented:
I've configured the plugins to be signed as community plugins.

Please consider providing a test configuration using the following template:

https://github.com/grafana/plugin-test-environment

petethepig commented:
Thank you @marcusolsson! I was able to sign the plugins so I updated this PR with signed versions.

I also set up the test environment in this repository: https://github.com/pyroscope-io/pyroscope-plugin-test-environment

Let me know if you need any help testing the plugins.

marcusolsson commented:
I'm getting an error sometimes when I change the Application in the query editor. I haven't found a way to reliably reproduce it, but if I keep changing the Application value, eventually it pops up.

<img width="137" alt="Screenshot 2021-06-11 at 13 37 55@2x" src="https://user-images.githubusercontent.com/8396880/121680802-42e58d00-caba-11eb-96c8-7739a4d9dc88.png">

I see that the data source is simply passing a single field-single row with a JSON body. This couples the panel and datasource so that they in practice can't be used with other plugins. I hope that you will consider finding a data frame schema that let users use the flame graph panel along with other data sources. 

Let me know if you want to look at the `s is undefined` before I publish this or if you want to fix it in the next release!

petethepig commented:
Hi @marcusolsson 

RE the error — we fixed the error, re-uploaded the plugin and updated this PR. I think it's good to go!

Regarding the coupling of the panel and datasource. This is a tricky problem — pyroscope returns data in a format that represents a tree. Other visualizations mostly work with arrays of some sort. So it's hard to map that tree data to arrays, while ensuring that the data visualized is useful. Having said that, we definitely see the appeal of being able to visualize pyroscope data with other visualizations. We'll think about it some more and we will probably add something in the next release.

marcusolsson commented:
I totally understand the challenge and it's not a blocker for publishing this. Something to keep at the back of your mind 😄 

I've published both plugins, but allow a few minutes before it's available.
